rpc/server.go

```diff
@@ -115,13 +116,19 @@ func (s *Server) handleUpdate(msg *Message) {
 
 func (s *Server) createResponse(count int, msg *Message, err error) {
 	msg.MagicVersion = MagicVersion
+	msg.Size = uint32(len(msg.Data))
+	if msg.Type == TypeWrite {
+		msg.Data = nil
+	}
 	msg.Type = TypeResponse
 	if err == io.EOF {
-		msg.Data = msg.Data[:count]
 		msg.Type = TypeEOF
+		msg.Data = msg.Data[:count]
```

Comment:
in case msg.Type == TypeWrite , msg.Data will be nil. can we do msg.Data[:count] in this case? Probably we should put a chekck if it's not TypeWrite then only set msg.data.